src/data_preprocessing.py

```
@@ -0,0 +1,211 @@
+    first_image_name = image_files[0]
+    img_path = os.path.join(sample_folder, first_image_name)
+
+    os.makedirs(output_vis_dir, exist_ok=True)
+    # Clean up previous visualizations
+    for f in os.listdir(output_vis_dir):
+        try:
+            os.path.join(output_vis_dir, f) # Logic check, not actual removal here to be safe?
+            # actually let's remove it as in original code
+            os.remove(os.path.join(output_vis_dir, f))
+        except Exception:
+            pass
```

Ruff (0.14.14)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)